src/components/Modals/ContactCardModal.spec.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react'
+import { ContactCardModal, DumbContactCardModal } from './ContactCardModal'
+import { render } from '@testing-library/react'
+import { createMockClient } from 'cozy-client'
+import AppLike from '../../tests/Applike'
+import { fakedContact } from '../../../fixtures/contacts'
+import { fakedAllGroups } from '../../../fixtures/groups'
+
+const client = createMockClient({})
+const setup = ({
+  editMode,
+  contact,
+  allGroups,
+  shouldDisplayConfirmDeleteModal
+}) => {
+  return {
+    editMode,
+    contact,
+    allGroups,
+    t: x => x,
+    toggleConfirmDeleteModal: jest.fn,
+    toggleEditMode: jest.fn,
+    shouldDisplayConfirmDeleteModal,
+    deleteContact: jest.fn
+  }
+}
+
+describe('ContactCardModal', () => {
+  it('should display the spinner', async () => {
+    const props = {
+      id: 'contactCardModal',
+      onClose: jest.fn,
+      deleteContact: jest.fn
+    }
+    const jsx = (
+      <AppLike client={client}>
+        <ContactCardModal {...props} />
+      </AppLike>
+    )
+
+    const { findByTestId } = render(jsx)
+    expect(await findByTestId('contactSpinner'))
+  })
+
+  it('should display the contact card', () => {
+    const props = setup({
+      editMode: false,
+      contact: fakedContact,
+      allGroups: fakedAllGroups,
+      shouldDisplayConfirmDeleteModal: false
+    })
+
+    const jsx = (
+      <AppLike client={client}>
+        <DumbContactCardModal {...props} />
+      </AppLike>
+    )
+    const { getByText } = render(jsx)
+    expect(getByText(fakedContact.fullname))
+  })
+
+  it('should display the delete confirmation modal', () => {
+    const props = setup({
+      editMode: false,
+      contact: fakedContact,
+      allGroups: fakedAllGroups,
+      shouldDisplayConfirmDeleteModal: true
+    })
+
+    const jsx = (
+      <AppLike client={client}>
+        <DumbContactCardModal {...props} />
+      </AppLike>
+    )
+
+    const { getByText } = render(jsx)
+    expect(getByText('delete-confirmation.title'))
```

Comment:
Nit: here we could also test that the other modal is also displayed (the one with the contact information).

Reply:
Done here https://github.com/cozy/cozy-contacts/pull/628/commits/0fce4c4a5a737a1f9969d5090ec51f6a6c0e3a7a